presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/DetermineJoinDistributionType.java

```diff
@@ -27,13 +28,10 @@
 import java.util.Map;
 import java.util.Optional;
 
-import static com.facebook.presto.SystemSessionProperties.isDistributedJoinEnabled;
-import static com.facebook.presto.sql.planner.optimizations.QueryCardinalityUtil.isAtMostScalar;
-import static com.facebook.presto.sql.planner.plan.JoinNode.Type.FULL;
-import static com.facebook.presto.sql.planner.plan.JoinNode.Type.INNER;
-import static com.facebook.presto.sql.planner.plan.JoinNode.Type.RIGHT;
+import static com.facebook.presto.sql.planner.iterative.Lookup.noLookup;
 import static java.util.Objects.requireNonNull;
 
+@Deprecated
```

Comment:
This hasn't the same functionality as `iterative/rule/DetermineJoinDistributionType.java` (it handles delete node). Consider either refactoring this class to `ReplicateJoinUnderDelete` to be consistent with semi join. I don't see much value in keeping this class.

Reply:
It is equivalent but keeping it around allows disabling the new rule.